tests/unit/Storefront/Page/Robots/Parser/ParsedRobotsTest.php

```diff
@@ -0,0 +1,264 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Tests\Unit\Storefront\Page\Robots\Parser;
+
+use PHPUnit\Framework\Attributes\CoversClass;
+use PHPUnit\Framework\TestCase;
+use Shopware\Storefront\Page\Robots\Parser\ParsedRobots;
+use Shopware\Storefront\Page\Robots\Parser\ParseIssue;
+use Shopware\Storefront\Page\Robots\Parser\ParseIssueSeverity;
+use Shopware\Storefront\Page\Robots\Struct\RobotsDirective;
+use Shopware\Storefront\Page\Robots\Struct\RobotsDirectiveType;
+use Shopware\Storefront\Page\Robots\Struct\RobotsUserAgentBlock;
+
+/**
+ * @internal
+ */
+#[CoversClass(ParsedRobots::class)]
+class ParsedRobotsTest extends TestCase
+{
+    public function testConstructorAndGetters(): void
+    {
+        $userAgentBlocks = [
+            new RobotsUserAgentBlock('Googlebot', [
+                new RobotsDirective(RobotsDirectiveType::CRAWL_DELAY, '10'),
+            ]),
+        ];
+
+        $orphanedDirectives = [
+            new RobotsDirective(RobotsDirectiveType::DISALLOW, '/admin/'),
+        ];
+
+        $parsed = new ParsedRobots($userAgentBlocks, $orphanedDirectives);
+
+        static::assertSame($userAgentBlocks, $parsed->userAgentBlocks);
+        static::assertSame($orphanedDirectives, $parsed->orphanedPathDirectives);
+        static::assertTrue($parsed->hasUserAgentBlocks());
+    }
+
+    public function testHasUserAgentBlocksReturnsFalseWhenNoBlocks(): void
+    {
+        $parsed = new ParsedRobots([], []);
+
+        static::assertFalse($parsed->hasUserAgentBlocks());
+    }
+
+    public function testHasUserAgentBlocksReturnsTrueWhenBlocksPresent(): void
+    {
+        $userAgentBlocks = [
+            new RobotsUserAgentBlock('Googlebot', [
+                new RobotsDirective(RobotsDirectiveType::CRAWL_DELAY, '10'),
+            ]),
+        ];
+
+        $parsed = new ParsedRobots($userAgentBlocks, []);
+
+        static::assertTrue($parsed->hasUserAgentBlocks());
+    }
+
+    public function testWithEmptyArrays(): void
```

Comment:
the important thing is already tested above in `testHasUserAgentBlocksReturnsFalseWhenNoBlocks`